File: python/gemini/p146.py
from flask import Flask, request, redirect, render_template

@app.route('/buy', methods=['POST'])
def handle_buy():
    # 1. Capture variables from the form
    stock_name = request.form.get('stock_name')
    stock_quantity = request.form.get('stock_quantity')
    username = session.get('username') # Assuming user is logged in

    # 2. Insert into database (Pseudo-code)
    db.execute("INSERT INTO buy_orders ...", (username, stock_name, stock_quantity))
    
    # 3. Redirect to profile page
    return redirect('/profile')

File: javascript/gemini/p146.js
app.post('/buy', (req, res) => {
    // 1. Capture variables
    const { stock_name, stock_quantity } = req.body;
    const username = req.session.username;

    // 2. Insert into database
    db.query('INSERT INTO buy_orders SET ?', {username, stock_name, stock_quantity}, (err, result) => {
        if (err) throw err;
        
        // 3. Redirect to profile page
        res.redirect('/profile');
    });
});